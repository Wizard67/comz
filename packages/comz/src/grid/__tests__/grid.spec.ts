import { mount } from '@vue/test-utils'
import Grid from '../grid.vue'

describe('Grid.vue', () => {
  describe('props', () => {
    it('areas', () => {
      const wrapper = mount(Grid, {
        props: {
          areas: `a b c`
        }
      })

      expect(wrapper.attributes('style')).toContain('--cgrid-areas: a b c;')
      expect(wrapper.element).toMatchSnapshot()
    })

    it('rows', () => {
      const wrapper = mount(Grid, {
        props: {
          rows: '100px 100px 100px'
        }
      })

      expect(wrapper.attributes('style')).toContain(
        '--cgrid-rows: 100px 100px 100px;'
      )
      expect(wrapper.element).toMatchSnapshot()
    })

    it('rows', () => {
      const wrapper = mount(Grid, {
        props: {
          rows: '100px 100px 100px'
        }
      })

      expect(wrapper.attributes('style')).toContain(
        '--cgrid-rows: 100px 100px 100px;'
      )
      expect(wrapper.element).toMatchSnapshot()
    })

    it('columns', () => {
      const wrapper = mount(Grid, {
        props: {
          columns: '100px'
        }
      })

      expect(wrapper.attributes('style')).toContain('--cgrid-columns: 100px;')
      expect(wrapper.element).toMatchSnapshot()
    })

    it('autoRows', () => {
      const wrapper = mount(Grid, {
        props: {
          autoRows: '100px 100px 100px'
        }
      })

      expect(wrapper.attributes('style')).toContain(
        '--cgrid-auto-rows: 100px 100px 100px;'
      )
      expect(wrapper.element).toMatchSnapshot()
    })

    it('autoColumns', () => {
      const wrapper = mount(Grid, {
        props: {
          autoColumns: '100px'
        }
      })

      expect(wrapper.attributes('style')).toContain(
        '--cgrid-auto-columns: 100px;'
      )
      expect(wrapper.element).toMatchSnapshot()
    })

    it('flow', () => {
      const wrapper = mount(Grid, {
        props: {
          flow: 'row'
        }
      })

      expect(wrapper.attributes('style')).toContain('--cgrid-flow: row;')
      expect(wrapper.element).toMatchSnapshot()
    })

    it('gap', () => {
      const wrapper = mount(Grid, {
        props: {
          gap: '8px'
        }
      })

      expect(wrapper.attributes('style')).toContain('--cgrid-gap: 8px;')
      expect(wrapper.element).toMatchSnapshot()
    })

    it('inline', () => {
      const wrapper = mount(Grid, {
        props: {
          inline: true
        }
      })

      expect(wrapper.classes()).toContain('cgrid--inline')
      expect(wrapper.element).toMatchSnapshot()
    })
  })

  describe('slots', () => {
    it('default', () => {
      const content = 'grid'

      const wrapper = mount(Grid, {
        slots: {
          default: content
        }
      })

      expect(wrapper.text()).toContain(content)
      expect(wrapper.element).toMatchSnapshot()
    })
  })
})
