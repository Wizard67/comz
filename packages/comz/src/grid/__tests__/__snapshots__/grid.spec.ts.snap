// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Grid.vue props areas 1`] = `
<div
  class="cgrid"
  style="--cgrid-areas: a b c; --cgrid-rows: initial; --cgrid-columns: initial; --cgrid-auto-rows: initial; --cgrid-auto-columns: initial; --cgrid-flow: initial; --cgrid-gap: initial;"
>
  
  
</div>
`;

exports[`Grid.vue props autoColumns 1`] = `
<div
  class="cgrid"
  style="--cgrid-areas: initial; --cgrid-rows: initial; --cgrid-columns: initial; --cgrid-auto-rows: initial; --cgrid-auto-columns: 100px; --cgrid-flow: initial; --cgrid-gap: initial;"
>
  
  
</div>
`;

exports[`Grid.vue props autoRows 1`] = `
<div
  class="cgrid"
  style="--cgrid-areas: initial; --cgrid-rows: initial; --cgrid-columns: initial; --cgrid-auto-rows: 100px 100px 100px; --cgrid-auto-columns: initial; --cgrid-flow: initial; --cgrid-gap: initial;"
>
  
  
</div>
`;

exports[`Grid.vue props columns 1`] = `
<div
  class="cgrid"
  style="--cgrid-areas: initial; --cgrid-rows: initial; --cgrid-columns: 100px; --cgrid-auto-rows: initial; --cgrid-auto-columns: initial; --cgrid-flow: initial; --cgrid-gap: initial;"
>
  
  
</div>
`;

exports[`Grid.vue props flow 1`] = `
<div
  class="cgrid"
  style="--cgrid-areas: initial; --cgrid-rows: initial; --cgrid-columns: initial; --cgrid-auto-rows: initial; --cgrid-auto-columns: initial; --cgrid-flow: row; --cgrid-gap: initial;"
>
  
  
</div>
`;

exports[`Grid.vue props gap 1`] = `
<div
  class="cgrid"
  style="--cgrid-areas: initial; --cgrid-rows: initial; --cgrid-columns: initial; --cgrid-auto-rows: initial; --cgrid-auto-columns: initial; --cgrid-flow: initial; --cgrid-gap: 8px;"
>
  
  
</div>
`;

exports[`Grid.vue props inline 1`] = `
<div
  class="cgrid cgrid--inline"
  style="--cgrid-areas: initial; --cgrid-rows: initial; --cgrid-columns: initial; --cgrid-auto-rows: initial; --cgrid-auto-columns: initial; --cgrid-flow: initial; --cgrid-gap: initial;"
>
  
  
</div>
`;

exports[`Grid.vue props rows 1`] = `
<div
  class="cgrid"
  style="--cgrid-areas: initial; --cgrid-rows: 100px 100px 100px; --cgrid-columns: initial; --cgrid-auto-rows: initial; --cgrid-auto-columns: initial; --cgrid-flow: initial; --cgrid-gap: initial;"
>
  
  
</div>
`;

exports[`Grid.vue props rows 2`] = `
<div
  class="cgrid"
  style="--cgrid-areas: initial; --cgrid-rows: 100px 100px 100px; --cgrid-columns: initial; --cgrid-auto-rows: initial; --cgrid-auto-columns: initial; --cgrid-flow: initial; --cgrid-gap: initial;"
>
  
  
</div>
`;

exports[`Grid.vue slots default 1`] = `
<div
  class="cgrid"
  style="--cgrid-areas: initial; --cgrid-rows: initial; --cgrid-columns: initial; --cgrid-auto-rows: initial; --cgrid-auto-columns: initial; --cgrid-flow: initial; --cgrid-gap: initial;"
>
  
  grid
  
</div>
`;
